sr_port/tcp_open.c

```diff
@@ -173,28 +125,30 @@ int tcp_open(char *host, unsigned short port, int4 timeout, boolean_t passive) /
 		if (-1 == setsockopt(lsock, SOL_SOCKET, SO_REUSEADDR, &on, SIZEOF(on)))
 		{
 			save_errno = errno;
-			(void)close(lsock);
+			CLOSEFILE(lsock, rc);
 			errptr = (char *)STRERROR(save_errno);
          		errlen = STRLEN(errptr);
                 	gtm_putmsg_csa(CSA_ARG(NULL) VARLSTCNT(7) ERR_SETSOCKOPTERR, 5,
-					LEN_AND_LIT("SO_REUSEADDR"), save_errno, errlen, errptr);
+				       LEN_AND_LIT("SO_REUSEADDR"), save_errno, errlen, errptr);
+			freeaddrinfo(ai_ptr);
```

Comment:
From the man page for getaddrinfo:

The  getaddrinfo()  function  allocates and initializes a linked list of addrinfo structures, one for each network address that matches node and service, subject to any restrictions imposed by hints, and returns a pointer to  the  start  of  the list in res.  The items in the linked list are linked by the ai_next field.

A 'block chain' is quite simply a chain of blocks (structures) - linked list  vs block chain - not much semantic difference but since it seems to be confusing I'll change it to linked list.